torch/_dynamo/guards.py

```diff
@@ -468,11 +468,22 @@ def TENSOR_MATCH(self, guard: Guard):
                 for term in terms:
                     real_value = self.get(tensor_name + "." + term)
                     code.append(f"{tensor_name}.{term} == {real_value}")
-                self._produce_guard_code(guard, code)
             else:
                 self.tensor_check_names.append(tensor_name)
                 self.tensor_check_examples.append(value)
 
+            if hasattr(value, "_dynamo_dynamic_indices"):
+                code.append(
+                    f"(not {tensor_name}._dynamo_dynamic_indices.isdisjoint({value._dynamo_dynamic_indices})) if hasattr({tensor_name}, '_dynamo_dynamic_indices') else True"
+                )
+            else:
+                code.append(
+                    f"hasattr({tensor_name}, '_dynamo_dynamic_indices') == False"
```

Comment:
OK, we've flip flopped enough on this that you are going to need to put a comment here.

I am still not sure this is the guard code you want. Right now, the logic says:

1. If we compiled this code with a marked tensor
  a. If you call the compiled code with a non-marked tensor, REUSE
  b. But if you call the compiled code with a marked tensor,
    1. If the new marked tensor's dynamic dims are completely disjoint from the original marked tensor dims, RECOMPILE
    2. Otherwise, REUSE
2. Otherwise (we compiled this code without a marked tensor)
  a. If you call the compiled code with a non-marked tensor, REUSE
  b. Otherwise, RECOMPILE

Branch (2) seems fine. But Branch (1) seems wrong. Specifically, disjointness seems like the wrong thing to test. Don't you want some sort of non-strict inclusion.

Reply:
I thought disjoint did not strict inclusion, but I am wrong here, what we really want is "The new tensors dynamic dims must be in set {old_tensor_dynamic_dims}"

Reply:
I think {new_set} - {old_set} == {empty_set} should suffice?